docs/advanced/test-command.md

```diff
@@ -0,0 +1,91 @@
+---
+sidebar_position: 1
+description: Test networking between Charon dependant services
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Network tests
+
+:::caution
+The charon test command is in alpha state and should be used with caution on testnets only.
+:::
+
+Charon test command is evaluates effectiveness of the network between the machine it is running on and the targeted external service - other Charon peers, beacon node(s) or validator client. It prints a performance report to the standard output (can be omitted by with `quiet` flag) and a machine-readable TOML format of the report if `output-toml` flag is set.
+
+## Peers
+
+Run tests towards other Charon peers to evaluate the effectiveness of a potential cluster setup. The command sets up a libp2p node, similarly to what Charon originally does. This test command **has to be running simultanuously with other peers**. After the node is up it has some await time for other peers to get their nodes up and running, retrying every 3 seconds. The libp2p node connects to relays (configurable with `p2p-relays` flag) and to other libp2p nodes via TCP. Other peer nodes are discoverable by using their ENRs. Note that for a peer to be successfully discovered, it needs to be connected to the same relay. After completion of the test suite the libp2p node stays alive (duration configurable with `keep-alive` flag) for other peers to continue testing against it. The node can be forcefully stopped as well.
+
+To be able to establish direct connection, you have to ensure:
+
+- Your machine is publicly accessible on the internet or atleast a specific port is.
+- You add flag `p2p-tcp-address` (i.e.: `127.0.0.1:9001`) flag and the port specified in it is free and publicly accessible.
+- You add the flag `p2p-external-ip` (i.e.: `8.8.8.8`) and specify your public ip.
+
+If all points are satisfied by you and the other peers, you should be able to establish a direct TCP connection between each other. Note that a relay is still required, as establishing direct connection is facilitated by it.
+
+### Pre-requisites
+
+- [Create ENR](http://localhost:3000/docs/next/charon/charon-cli-reference#creating-an-enr-for-charon).
```

Comment:
Don't specify localhost here. 

@OisinKyne can we do relative linking in Docusaurus somehow?

Reply:
Ouf, missed this one, thanks! Added relative link by `../charon/charon-cli-reference#creating-an-enr-for-charon`